### ISSUE TITLE
ITKImageReader not working properly

### ISSUE THREAD

**Issue by katiamar63** [InsightSoftwareConsortium/itk-js#128]
I use ITK Readers for loading MRI images (nii, mnc, vtk etc.) . In order to not re-invent the wheel I am using Reader Factory approach  based on VTK and ParaView Glance. I have noticed that resources (probably of workers) are not released on the ITK side. This happens when I want  to read a given resource (file or url) twice in the same browser session. I thought that maybe this is only case of my application, but I have checked in ParaView and got the same result.    I made the following experiment in PV:
- loaded nrrd file  (about 500KB)
- loaded vtk file (about 800KB)
- again the same nrrd.

Every time I had exceptions (sometimes different). 
![ioworker1](https://user-images.githubusercontent.com/3754848/39558891-577a165c-4e60-11e8-94d9-03a2782fdb2a.PNG)
![ioworker2](https://user-images.githubusercontent.com/3754848/39558941-a33bb71c-4e60-11e8-8ba6-9ce5734d2ca9.PNG)

In case of my application  I have  binding error related to unknown  emsc parameter (see below). How can i solve this problem? I want to give to the user possibility to load the same image as many times as he wants during session with browser without refreshing page. Should  I terminate workers, clear readimagearraybuffer, and if so, how can I do it?  
![ioworker3](https://user-images.githubusercontent.com/3754848/39559090-b0541434-4e61-11e8-8752-68d09fce0861.PNG)

**Comment by floryst** [comment]
You should terminate the webworkers (best solution so far). This should be addressed in the most recent vtk-js ITKImageReader ([relevant commit](https://github.com/Kitware/vtk-js/commit/deb909505ca137f270752b6c99ba0eb8838796a4#diff-98b5cb072385ff859c1c529271298fd9R20)).

Side note: Glance stable should be updated soon, as that issue has already been fixed in Glance nightly.

**Comment by katiamar63** [comment]
Thanks for the tip. Please do not close this issue for a while,

**Comment by katiamar63** [comment]
Thanks again @floryst for help, but I need to ask you for a small tip, because I cannot make it work. 

When I debugged ITKImageReader in my application, I had noticed that readImageArrayBuffer had only three parameters. Because I was using  ITK v. 8.1.  I knew this was the issue why I cannot terminate worker. So I upgraded  ITK to 9.0 and I found that itk/readImageArrayBuffer  returns a Promise object that is resolved with the {image, webworker}, where webworker is some object, but image is 'undefined'.   

Concluding, after upgrade I could not load even single image (earlier I had problem when I was trying to load the same file two times).   What can be reason for that? I use vtk 6.4.22 with ITKImageReader as in your post, and when I try to use it with itk 9.0  ITKImageReader throws error in line 45 - itkImage is undefined.

I tracked that problem starts in itk/readImageBufferArray.js line 20
```
.then(function (image) {
            return Promise.resolve({ image, webWorker: worker })
        })
```
    where image is 'undefined'. What can be the reason? Can this be related to itkConfig?
